### PR TITLE
GRF-394 Add channel selector on category edit form

### PR DESCRIPTION
**Pull request by amigaire**
**Description (for Contributor and Core Developer)**

Add a channel selector on the Category edit form page.
It does not load locales linked to the selected channel yet.

**Definition Of Done (for Core Developer only)**

### CODE REVIEW

File: src/Akeneo/Category/front/src/feature/components/channel/ChannelSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                    }}
+                  >
+                    <HighlightChannel
+                      code={"["+channel.code+"]"}
+                      label={channel.labels['en_US']}
+                      selected={channel.code === value}
+                      key={channel.code}
+                    />
+                    {completeValues && !completeValues.includes(channel.code) && (
+                        <Pill level="warning" data-testid={`ChannelSelector.incomplete.${channel.code}`} />
```
Comment (BryanS29-akeneo) [inline src/Akeneo/Category/front/src/feature/components/channel/ChannelSelector.tsx:92]: why do we need the Pill component?

File: src/Akeneo/Category/front/src/feature/components/channel/ChannelSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                return (
+                  <ChannelDropdownItem
+                    aria-selected={channel.code === value}
+                    key={channel.code}
+                    onClick={() => {
+                      close();
+                      handleChange(channel.code);
+                    }}
+                  >
+                    <HighlightChannel
```
Comment (olivierpontier) [inline src/Akeneo/Category/front/src/feature/components/channel/ChannelSelector.tsx:85]: According to the DSM, the selected item should be handled by adding the property `isActive`: https://dsm.akeneo.com/?path=/docs/components-dropdown--simple-item. The style looks to be managed by the DSM also

File: src/Akeneo/Category/front/src/feature/components/channel/ChannelSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                      code={"["+channel.code+"]"}
+                      label={channel.labels['en_US']}
```
[the remark this answers is not in the document]
Comment (amigaire) [reply src/Akeneo/Category/front/src/feature/components/channel/ChannelSelector.tsx:87]: Will do. I did not implement it as it should be reworked to get locales list compatible with selected channel.

File: src/Akeneo/Category/front/src/feature/components/channel/ChannelSelector.tsx (hunk 2 above)
Comment (amigaire) [reply src/Akeneo/Category/front/src/feature/components/channel/ChannelSelector.tsx:85]: So much better when you don't have to rewrite the style :heart_eyes:

File: src/Akeneo/Category/front/src/feature/components/channel/ChannelSelector.tsx (hunk 1 above)
Comment (amigaire) [reply src/Akeneo/Category/front/src/feature/components/channel/ChannelSelector.tsx]: not needed, removed

File: src/Akeneo/Category/front/src/feature/components/channel/ChannelSelector.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  SwitcherButton,
+  useBooleanState
+} from 'akeneo-design-system';
+import React, {forwardRef, Ref} from 'react';
+
+const ChannelSpan = forwardRef<HTMLSpanElement, ChannelProps>(
+    ({code, label, ...rest}: ChannelProps, forwardedRef: Ref<HTMLSpanElement>) => {
+      return (
+          <span ref={forwardedRef} {...rest}>
+          {label || "["+code+"]"}
```
Comment (BryanS29-akeneo) [inline src/Akeneo/Category/front/src/feature/components/channel/ChannelSelector.tsx:13]: you can replace your string concatantion by this: 
```
{label || `[${code}]`}
```

File: src/Akeneo/Category/front/src/feature/components/channel/ChannelSelector.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  const catalogLocale = userContext.get('catalogLocale');
+  const [isOpen, open, close] = useBooleanState(false);
+  const selectedChannel: Channel = values.find(channel => channel.code === value) || values[0];
+
+  const handleChange = (channelCode: ChannelCode) => onChange?.(channelCode);
+
+  return (
+    <Dropdown>
+      <SwitcherButton inline onClick={() => {open();}} label={translate('pim_common.channel')}>
+        <ChannelSpan code={"["+selectedChannel.code+"]"} label={selectedChannel.labels[catalogLocale]} />
```
Comment (BryanS29-akeneo) [inline src/Akeneo/Category/front/src/feature/components/channel/ChannelSelector.tsx:41]: you can replace your string concatantion by this: 
```
code=`[${selectedChannel.code}]`
```

**Review by BryanS29-akeneo — APPROVED** [review]

**Review by jeremyBeaucousinAkeneo — COMMENTED** [review]
Some exemple of string interpolation in typescript: https://bobbyhadz.com/blog/typescript-string-interpolation

**Review by jeremyBeaucousinAkeneo — APPROVED** [review]

File: src/Akeneo/Category/front/src/feature/components/EditAttributesForm.tsx [hunk 6]
```diff
@@ -129,6 +132,14 @@ export const EditAttributesForm = ({attributeValues, template, onAttributeValueC
       <SectionTitle>
         <SectionTitle.Title>{translate('Attributes')}</SectionTitle.Title>
         <SectionTitle.Spacer />
+        <ChannelSelector
+            value={channel}
+            values={Object.values(channels)}
```
[the remark this answers is not in the document]
Comment (amigaire) [reply src/Akeneo/Category/front/src/feature/components/EditAttributesForm.tsx:137]: Be cause channels contains a list of Channel type. If I am wrong, ping me to discuss about it

File: src/Akeneo/Category/front/src/feature/components/EditAttributesForm.tsx [hunk 7]
```diff
@@ -129,6 +132,14 @@ export const EditAttributesForm = ({attributeValues, template, onAttributeValueC
...
+            onChange={value => {
+              setChannel(value);
+              userContext.set('catalogChannel', value, {});
+            }}
```
[the remark this answers is not in the document]
Comment (amigaire) [reply src/Akeneo/Category/front/src/feature/components/EditAttributesForm.tsx:146]: It's cleaner indeed, I changed it

**Review by olivierpontier — APPROVED** [review]

**Pull request merged**